beekeeper-path-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/handler/MessageEventHandler.java

```diff
@@ -53,23 +50,25 @@
     this.lifecycleEventType = lifecycleEventType;
   }
 
-  public List<HousekeepingPath> handleMessage(MessageEvent event) {
+  public List<Housekeeping> handleMessage(MessageEvent event) {
     ListenerEvent listenerEvent = event.getEvent();
 
     if (shouldFilterMessage(listenerEvent)) {
       return Collections.emptyList();
     }
 
-    return generateHouseKeepingPaths(listenerEvent);
+    return generateHousekeepingEntities(listenerEvent);
   }
 
-  abstract List<EventModel> generateEventModels(ListenerEvent listenerEvent);
-
   protected abstract List<ListenerEventFilter> getFilters();
```

Comment:
This could perhaps be set in the constructor?

Reply:
Yeah we could but then it makes a bit mess when extend the class because the `super()` method has to be called first in the parameter. So creating the list instance in the super call will make it a bit messy. Example:

This doesn't work
```
  @Autowired
  public ExpiredMessageHandler(
      @Value("${properties.beekeeper.default-expiration-delay}") String cleanupDelay
  ) {
    List<ListenerEventFilter> filters = List.of(
        new EventTypeListenerEventFilter(EVENT_CLASSES),
        new TableParameterListenerEventFilter());
    super(cleanupDelay, EXPIRED_DATA_RETENTION_PERIOD_PROPERTY_KEY, LIFECYCLE_EVENT_TYPE, filters);
  }
```

This will get messy if you have lots of filters:
```
  @Autowired
  public ExpiredMessageHandler(
      @Value("${properties.beekeeper.default-expiration-delay}") String cleanupDelay
  ) {
    super(cleanupDelay, EXPIRED_DATA_RETENTION_PERIOD_PROPERTY_KEY, LIFECYCLE_EVENT_TYPE, List.of(
        new EventTypeListenerEventFilter(EVENT_CLASSES),
        new TableParameterListenerEventFilter()));
  }
```

What do you think?

Reply:
```
private final static List<> FILTERS = List.of(
        new EventTypeListenerEventFilter(EVENT_CLASSES),
        new TableParameterListenerEventFilter());

@Autowired
  public ExpiredMessageHandler(
      @Value("${properties.beekeeper.default-expiration-delay}") String cleanupDelay
  ) {
    super(cleanupDelay, EXPIRED_DATA_RETENTION_PERIOD_PROPERTY_KEY, LIFECYCLE_EVENT_TYPE, FILTERS);
  }
```

maybe?

Reply:
Not sure if that would work because we insert mock filter objects with this other constructor for tests:

```
  public ExpiredMessageHandler(
      @Value("${properties.beekeeper.default-expiration-delay}") String cleanupDelay,
      List<ListenerEventFilter> filters
  ) {
    super(cleanupDelay, EXPIRED_DATA_RETENTION_PERIOD_PROPERTY_KEY, LIFECYCLE_EVENT_TYPE);
    this.filters = filters;
  }
```

Any other ideas?